File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Cards/Status/Dazed.ts
import FileHelper from "../../FileHelper";
import { LocalizationProperty } from "../../Gen/DB/Localization";
import AbstractCard, { CardColor, CardRarity, CardTarget, CardType } from "../AbstractCard";

export class Dazed extends AbstractCard{
    Color: CardColor = CardColor.COLORLESS;
    Desc: string = LocalizationProperty.Read("卡牌-晕眩描述");
    Icon: string = FileHelper.FormatPath("cards/status/dazed.png");
    Name: string = LocalizationProperty.Read("卡牌-晕眩");
    Rarity: CardRarity = CardRarity.COMMON;
    Target: CardTarget = CardTarget.NONE;
    Type: CardType = CardType.STATUS;
    Ethereal = true;
}

File: Unity/Assets/Scripts/ModTheSpire/Card/Status/Dazed.cs
using TCG.Config;

namespace ModTheSpire.Card.Status
{
    public class Dazed : AbstractCard
    {
        public override string Desc { get; } = LocalizationProperty.Read("卡牌-晕眩描述");
        public override string Icon { get; } = "cards/status/dazed.png";
        public override CardType Type { get; } = CardType.STATUS;
        public override CardRarity Rarity { get; } = CardRarity.COMMON;
        public override CardTarget Target { get; } = CardTarget.NONE;
        public override CardColor Color { get; } = CardColor.COLORLESS;

        public Dazed()
        {
            Ethereal = true;
        }
    }
}